mirror/github/download.py

```diff
@@ -0,0 +1,75 @@
+import httpx
+import click
+import pygit2
+import os
+import json
+import traceback
+from pathlib import Path
+
+
+
+@click.command(context_settings=dict(help_option_names=['-h', '--help']))
+@click.option('--output_directory', '-d', default='.', help='Output dir for repos.', show_default=True)
+@click.option('--stars_condition', '-s', default='>500', help='Stars search condition.', show_default=True)
+@click.option('--languages', '-ls', default='[python]', help='--languages "[o11, o12, o13]" or [o11,o12,o13]', show_default=True)
+@click.option('--token', '-t', help='Access token for increase rate limit. Read from $github_token if specify.', default='', show_default=True)
+@click.option('--amount', '-n', help='Amount of repo.', default=50, show_default=True)
+def repos(output_directory,stars_condition,languages,token, amount):
+    token= os.environ.get('github_token', token)
+
+    
+
+    if token == '':
+        click.echo(f'Token not found!')
+        raise
+
+    
+    headers = {'accept': 'application/vnd.github.v3+json',
+                'Authorization': f'token {token}'}
+
+    try:
+        languages = json.loads(languages)
+        
+    except ValueError:
+        pass
+    
+
+    resolve_path = Path(output_directory)
+
+    if not os.path.exists(resolve_path):
+        os.makedirs(resolve_path)
+
+    languages = languages[1:-1]  # trim '[' and ']'
```

Comment:
Don't write your own logic for this. `click` already has variadic arguments:
https://click.palletsprojects.com/en/7.x/arguments/#variadic-arguments